### PR TITLE
Add tree string

### PR DESCRIPTION
**Pull request by jeffbrennan**
addresses #88

See comment below

### CODE REVIEW

**Comment by jeffbrennan** [comment]
hey @MrPowers, this is ready for review. I based the output on how pyspark prints schemas with the `printSchema()` function.  Verified that this approach works with multiple levels of nested structs, schemas of different lengths, ignore_nullable T/F, and ignore_metadata T/F

Standard example: 
image

Different lengths: 
image

Multiple nested structs:
image

---

Some thoughts for additional features

- Currently this auto-determines whether the output should be a table or a tree string based on whether the provided schemas have nested structs or > 10 columns. Should this be user-configurable? 
- Truncation of wide tree string outputs. If a tree string line is wider than a certain threshold (~80 characters), should I just truncate the right schema output, or both the left and right evenly? 
- Output looks good in the terminal but not in VSCode when an exception is caught. Should we just have a standard `SchemasNotEqualError` message and log the diff tree/table separately in the terminal? 
image

- ignore_metadata = True is a little confusing currently (see name column in example below). Should we add a text indicator to explain that two lines are differing because of their metadata? 
image

**Review by SemyonSinchenko — APPROVED** [review]
LGTM

**Comment by kleemaj** [comment]
This fix is super helpful! Any chance it can get merged? Happy to try resolving the merge conflicts if you need help.

**Comment by jeffbrennan** [comment]
> This fix is super helpful! Any chance it can get merged? Happy to try resolving the merge conflicts if you need help.

thanks for putting this back on my radar! I'll work on resolving the merge conflicts

**Comment by jeffbrennan** [comment]
@fpgmaas @SemyonSinchenko  could one of y'all please review the changes after my recent updates?

For the merge conflicts, most of the differences were formatting/line length related and I took the incoming version from main in those instances. 

I also experienced some pytest failures after the merge that I have addressed in the subsequent commits.

- reference to the `six` package in `schema_comparer.py` but not in `pyproject.toml` or the imports at the top of the file
- missing import of `create_schema_comparison_tree` in `test_schema_comparer.py`
- failure on passing a structfield to the `blue()` function
- missing bcolors imports in `schema_comparer.py`

Let me know if anything should be changed!

File: chispa/schema_comparer.py [hunk 1]
```diff
@@ -3,9 +3,11 @@
 import typing
 from itertools import zip_longest
 
+import six
```
Comment (fpgmaas) [inline chispa/schema_comparer.py:6]: I don't think we need six; We ca use [itertools.zip_longest](https://github.com/MrPowers/chispa/blob/cccff0784017ab0b235951af5e21cfc196d1b0dc/chispa/rows_comparer.py#L3) from the standard library.

Comment (jeffbrennan) [reply chispa/schema_comparer.py]: updated!

**Comment by jeffbrennan** [comment]
> @jeffbrennan build is failing due to formatting issues. Could you run the pre-commit hooks locally? Sorry, we should reflect this in `CONTRIBUTING.md`

passing locally after my recent changes
image

**Comment by jeffbrennan** [comment]
hi @fpgmaas, pinging you again to see if we can get this merged

**Pull request merged**